### ISSUE TITLE
GPG signature by default

### ISSUE THREAD

**Issue by aonez** [ForkIssues/Tracker#271]
Would be amazing to set a "global" signature in Fork, so any repository uses it by default. Similar to the credentials "Use global git credentials" option.

**Comment by kervich** [comment]
I would disagree. While I suppose that a git client may set an option to use a certain GPG key for a repository that is currently open in it, it should never make assumptions about any other repositories. Reading settings from a global config file is ok (unless those are never modified by the client), and while it may be ok for credentials, I'm not sure it is ok for a signing GPG key (just because some user may have mistakenly set `commit.gpgSign` in system config, that key should not be used in every repository in the system. Same with global (user) gitconfig - you might need to make a commit in an organisation repo at some point and signing a commit there with your personal GPG key might be inappropriate), so I would expect a git client to only save the key id for a repository where I did change the setting to sign commits with some key (that's when I set a key when editing repo settings with that git client)

**Comment by aonez** [comment]
@kervich I agree this is more sensitive than the credentials. Anyway adding the option disabled by default should be useful, so the user must check "Use global GPG key".

**Comment by NanoSector** [comment]
I agree with having it as a global setting which is applied on new repositories only, seeing as the repositories already have a per-repo GPG setting.

**Comment by bwmravec** [comment]
Yes, having a default GPG signing for commits would be great, instead of setting it up every time for the new repo
